model/word2vec/state_test.go

```diff
@@ -18,43 +18,73 @@ import (
 	"bytes"
 	"io"
 	"io/ioutil"
+	"strings"
 	"testing"
 
+	"github.com/chewxy/gorgonia/tensor"
+	"github.com/chewxy/lingo/corpus"
 	"github.com/ynqa/word-embedding/config"
 	"github.com/ynqa/word-embedding/model"
 )
 
-var mockConfig = model.NewConfig(
-	config.DefaultLower,
-	config.DefaultDimension,
-	config.DefaultWindow,
-	config.DefaultInitLearningRate,
-)
+// MockOperator satisfies the interface of Optimizer.
+type MockOptimizer struct{}
+
+func (m *MockOptimizer) Init(c *corpus.Corpus, dimension int) error { return nil }
+func (m *MockOptimizer) Update(targetID int, contextVector, poolVector tensor.Tensor, learningRate float64) error {
+	return nil
+}
+
+// MockNopSeeker stores io.ReadCloser with Seek func that has nothing.
+type MockNopSeeker struct{ io.ReadCloser }
 
-var mockState = NewState(
-	mockConfig,
-	mockOpt,
-	config.DefaultSubsampleThreshold,
-	config.DefaultTheta,
-	config.DefaultBatchSize,
+func (n MockNopSeeker) Seek(offset int64, whence int) (int64, error) { return 0, nil }
+
+var (
+	testText   = "A B B C C C C"
+	testConfig = model.NewConfig(
+		config.DefaultLower,
+		config.DefaultDimension,
+		config.DefaultWindow,
+		config.DefaultInitLearningRate,
+	)
+	mockOpt       Optimizer = new(MockOptimizer)
+	mockNopSeeker           = MockNopSeeker{ReadCloser: ioutil.NopCloser(bytes.NewReader([]byte(testText)))}
 )
 
-var mockText = "A A A A B B B B C C C C"
+func testCorpus() *corpus.Corpus {
+	var emptyOpt corpus.ConsOpt = func(c *corpus.Corpus) error { return nil }
 
-type nopSeeker struct {
-	io.ReadCloser
+	c, _ := corpus.Construct(emptyOpt)
+	for _, word := range strings.Fields(testText) {
+		c.Add(word)
+	}
+	return c
 }
 
-func (n nopSeeker) Seek(offset int64, whence int) (int64, error) {
-	return 0, nil
+// testInitializedState returns *State has already called Preprocess.
+func testInitializedState() *State {
```

Comment:
this is poor naming - maybe `newTestState`? (using `test` as the first word of the name makes it sound like a verb)

Reply:
Sure. I'll fix it.